### PR TITLE
Test/additional guestbook dump

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:
Add additional dump for failing guestbook tests.
Also deprecated the old test framework

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: extensions/test/integration/operation/shoot_operation.go [hunk 1]
```diff
@@ -17,14 +17,14 @@ package operation
 import (
 	"context"
 	"fmt"
+	"github.com/gardener/gardener/test/framework"
```
Comment (timebertt) [inline extensions/test/integration/operation/shoot_operation.go]: Move down to 3rd party import block

File: test/framework/applications/guestbooktest.go [hunk 2]
```diff
@@ -191,8 +191,26 @@ func (t *GuestBookTest) Test(ctx context.Context) {
...
+	identifier := "[GUESTBOOK]"
+	err := t.framework.DumpDefaultResourcesInNamespace(ctx, identifier, t.framework.ShootClient, t.framework.Namespace)
+	if err != nil {
+		t.framework.Logger.Errorf("unable to dump guestbook resources in namespace %s: %s", t.framework.Namespace, err.Error())
+	}
+}
+
 // Cleanup cleans up all resources depoyed by the guestbook test
 func (t *GuestBookTest) Cleanup(ctx context.Context) {
+
```
Comment (timebertt) [inline test/framework/applications/guestbooktest.go]:
```suggestion
```

File: test/framework/k8s_utils.go [hunk 3]
```diff
@@ -3,6 +3,7 @@ package framework
 import (
 	"context"
 	"fmt"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
```
Comment (timebertt) [inline test/framework/k8s_utils.go]: Move down to same repo import block

File: extensions/test/integration/operation/shoot_operation.go [hunk 4]
```diff
@@ -70,7 +70,7 @@ func ScaleDeployment(setupContextTimeout time.Duration, client client.Client, de
 	ctxSetup, cancelCtxSetup := context.WithTimeout(context.Background(), setupContextTimeout)
```
Comment (timebertt) [inline extensions/test/integration/operation/shoot_operation.go]: Why is this `func` here? Can't we reuse `framework.ScaleDeployment()`?

Comment (timebertt) [reply extensions/test/integration/operation/shoot_operation.go]: Maybe we can also directly move the other function into `test/framework`...

File: test/framework/k8s_utils.go [hunk 5]
```diff
@@ -219,6 +220,66 @@ func DeleteResource(ctx context.Context, k8sClient kubernetes.Interface, resourc
...
+	if apierrors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve the replica count of the %s deployment: '%v'", name, err)
+	}
+	if replicas == nil || *replicas == *desiredReplicas {
+		return replicas, nil
+	}
+	// scale the scheduler deployment
```
Comment (timebertt) [inline test/framework/k8s_utils.go]:
```suggestion
	// scale the deployment
```

File: test/framework/k8s_utils.go [hunk 6]
```diff
@@ -219,6 +220,66 @@ func DeleteResource(ctx context.Context, k8sClient kubernetes.Interface, resourc
 	return err
 }
 
+// ScaleDeployment scales a deployment and waits until it is scaled
+func ScaleDeployment(setupContextTimeout time.Duration, client client.Client, desiredReplicas *int32, name, namespace string) (*int32, error) {
```
Comment (timebertt) [inline test/framework/k8s_utils.go]: This is not "setup-related", right?
```suggestion
func ScaleDeployment(timeout time.Duration, client client.Client, desiredReplicas *int32, name, namespace string) (*int32, error) {
```

File: test/framework/k8s_utils.go [hunk 7]
```diff
@@ -316,6 +377,23 @@ func NewClientFromServiceAccount(ctx context.Context, k8sClient kubernetes.Inter
...
+		if err := c.Client().Get(ctx, client.ObjectKey{Namespace: podNamespace, Name: podName}, pod); err != nil {
+			return retry.SevereError(err)
+		}
+		if !health.IsPodReady(pod) {
+			f.Logger.Infof("Waiting for %s to be ready!!", podName)
+			return retry.MinorError(fmt.Errorf(`pod "%s/%s" is not ready: %v`, podNamespace, podName, err))
+		}
+
+		return retry.Ok()
+
```
Comment (timebertt) [inline test/framework/k8s_utils.go]:
```suggestion
```

File: test/integration/shoots/care/shoot_care.go [hunk 8]
```diff
@@ -57,7 +55,7 @@ var _ = ginkgo.Describe("Shoot Care testing", func() {
 		defer func(ctx context.Context) {
 			if origReplicas != nil {
 				f.Logger.Infof("Test cleanup: Scale API Server to '%d' replicas again", *origReplicas)
-				origReplicas, err = framework.ScaleDeployment(timeout, f.SeedClient.Client(), origReplicas, gardencorev1beta1constants.DeploymentNameKubeAPIServer, f.ShootSeedNamespace())
+				origReplicas, err = testframework.ScaleDeployment(timeout, f.SeedClient.Client(), origReplicas, gardencorev1beta1constants.DeploymentNameKubeAPIServer, f.ShootSeedNamespace())
```
Comment (timebertt) [inline test/integration/shoots/care/shoot_care.go]: Can you remove the import alias to make it consistent with other tests?
```suggestion
				origReplicas, err = framework.ScaleDeployment(timeout, f.SeedClient.Client(), origReplicas, gardencorev1beta1constants.DeploymentNameKubeAPIServer, f.ShootSeedNamespace())
```

**Review by timebertt — COMMENTED** [review]
Nice, only nits :)

File: extensions/test/integration/operation/shoot_operation.go (hunk 4 above)
Comment (schrodit) [reply extensions/test/integration/operation/shoot_operation.go]: the function is already in the framework so i just removed the function in the extension test and use the framework function.

**Review by timebertt — DISMISSED** [review]
Thanks!
/lgtm now

BTW, why is `Gardener CI/CD` committer of your commits?

**Comment by schrodit** [comment]
Looks like something has changed my git config to the gardener ci user.. 🙈 
Changed it back now to my real user

**Comment by timebertt** [comment]
Can you recommit then, please?

**Comment by schrodit** [comment]
yes pushed now with my user

**Review by dguendisch — APPROVED** [review]

**Pull request merged**